packages/flutter_tools/lib/src/compile.dart

```diff
@@ -168,7 +207,17 @@ class KernelCompiler {
 
     if (extraFrontEndOptions != null)
       command.addAll(extraFrontEndOptions);
-    command.add(mainPath);
+
+    Uri mainUri;
+    if (packagesPath != null) {
+      mainUri = _PackageUriMapper.findUri(mainPath, packagesPath);
+    }
+    if (mainUri != null) {
+      command.addAll(<String>['--packages', packagesPath, mainUri.toString()]);
```

Comment:
Not sure why we only pass packagesPath if we were able to resolve mainUri? Conceptually it feels that .packages file is useful even if main file is identified via file path.

Reply:
done